httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/InternalAbstractHttpAsyncClient.java

```diff
@@ -122,6 +129,7 @@ void internalClose(final CloseMode closeMode) {
                 }
             }
         }
+        this.scheduledExecutorService.shutdown();
```

Comment:
Stupid question: Should this block indefinitely if there is hight low or requests take long? No strong opinion here.

Reply:
@michael-o The scheduler merely submits requests for execution, it does not actually execute them. However it probably makes sense to shut down the executor immediately.